### ISSUE TITLE
Query related to the parameters for uploading the raw files

### ISSUE THREAD

**Issue by jit15soni** [ContactEngineering/topobank#971]
Hi
I kindly request to clarify some queries regarding uploading the raw files of AFM topographies to the "contact.engineering" web-app. 
1.I have raw data files of .xyz format, which I am uploading by going to "Add Measurement" tab. The x, y and z values in the file are in units of metre and the scan area size is (2 micron x 2 micron). So, in screenshot 1 below, I entered the sizes x and y as 0.000002 and kept the height scale as 1.0 since z values are in units of meter. When I go to "Analyze this Measurement" tap and check roughness parameters, the roughness (Sq) value comes out of the order of 10^-7 metre (screenshot 2) which is quite large and different from what measured using a typical software "Gwyddion" to measure roughness of AFM topographies. 

![Screenshot 2023-03-28 195122_physical size](https://user-images.githubusercontent.com/125862502/228280991-edc04611-dbac-47ec-81e0-c350d8c63f13.jpg)
![Screenshot 2023-03-28 195541_roughness1](https://user-images.githubusercontent.com/125862502/228282518-9d947ca2-7735-4c41-b267-bc60fbf78ba4.jpg)

2. If I keep the height scale as 0.01 (screenshot 3), and analyze the measurement for roughness, then the Sq value comes out of the order of 10^-9 metre (screenshot 4), which is reasonable, but the magnitude (4.2999) occurs distinctly from what we obtain from Gwddion software. Below in screenshot 5,  roughness value is reported using the software (2.358 nm).

![Screenshot 2023-03-28 195656_physical size2](https://user-images.githubusercontent.com/125862502/228288668-0d77e061-5744-4585-8b55-a0b9caf5d5c9.jpg)
![Screenshot 2023-03-28 195938_roughness2](https://user-images.githubusercontent.com/125862502/228288715-4a670fcc-0a78-4f33-b2bc-21053e2c5c8b.jpg)
![Screenshot 2023-03-28 201348_gyddion roughness](https://user-images.githubusercontent.com/125862502/228288746-0021f98c-7b95-4427-a39d-72173f51df75.jpg)

I request here to clarify how to enter the parameters to allow the accurate analysis of the raw data.

Thanks

**Comment by jit15soni** [comment]
Hi Pastewka,
Thanks for responding. 
I've shared the files with you via email to lars.pastewka@imtek.uni-freiburg.de

**Comment by pastewka** [comment]
Hi @jit15soni - that data looks good on my end. I think you may have compared different channels in those files. The files contain in total 4 channels with height information, called "Z-Axis" and "Z-AxisSensor" in forward and backward mode. You can selected the data source in contact.engineering when uploading the topography:

![image](https://user-images.githubusercontent.com/2530391/232231897-5609bc4a-102a-44a8-a481-c5e5398fdad5.png)

Unfortunately, you can't at present change the channel once uploaded. (The channel is part of the surface definition.)

In Gwyddion, you can select the channel in the data browser:

![image](https://user-images.githubusercontent.com/2530391/232231979-e913942c-2faf-45f1-9a61-0f7af3c69b44.png)

Can you check whether this resolves you issue?

**Comment by pastewka** [comment]
Hi @jit15soni is this problem resolved?

**Comment by jit15soni** [comment]
Hi Prof. @pastewka 

I apologize for the delay in reply. The issue was raised because the website earlier did not support the .nid format of the file while uploading. So, I tried to convert the file into xyz format and upload it. But, with xyz file, I could not understand what to keep the height scale (I've discussed the problem with the height scale in the first post of this issue). That's why I shared files with you of both formats via email.  And, I think when you mention different channels, they appear only with nid file and not xyx file. In addition, I also think that there should not be different channels when I am working with xyz files since they only contain xyz position values, if I am not wrong. I guess the main issue remains same with xyz files that the measured roughness data are not reliable using xyz files if height scale not chosen properly.

![Screenshot 2023-04-23 184257_xyz vs nid](https://user-images.githubusercontent.com/125862502/233841845-a21a2638-8d83-4c91-a6e7-72ae4286756f.jpg)

Since the website is now supporting the nid file (for which I enquired in issue#966), I can work with nid files. The only issue I am facing now is that the website is not properly removing the tilt and curvature of a few nid files, due to which the measured roughness data occurs significantly different from those obtained using Gywddion software. (I've also reported this in issue#966).

**Comment by pastewka** [comment]
Hi @jit15soni - regarding title correction: Did you turn that on and it is giving you something different than Gwyddiob, or did you simply not find where to turn it on?

**Comment by jit15soni** [comment]
Hi Prof. @pastewka 
Do you mean tilt correction? If that is so, I had set the detrend mode as "remove curvature and tilt"; otherwise, I did not find this "title correction" to turn on. Would you please let me know what this is for and from where it should be turned on, if required?

**Comment by pastewka** [comment]
Yes that is what I meant. But you are saying it does not work?

**Comment by jit15soni** [comment]
Yes, after keeping the detrend mode as "remove curvature and tilt" and height scale as 0.01, the measured roughness data does not match with that obtained using Gwyddion software. I kept the height scale as 0.01 because with this parameter the measured roughness value comes in the order of 10^(-9) meter (nanometer). (Gywddion software also gives roughness data in nanometer orders.)

![Screenshot 2023-04-24 001050_similar parameters different files same Sq value](https://user-images.githubusercontent.com/125862502/233858721-b3ce8eac-b4e3-4fdd-b149-7779158c270e.jpg)

Also, I would like to bring to your notice which I recently observed that I uploaded different files with xyz formats and all the files gave same Sq (RMS height) value as 4.2999 x 10^(-9) meter. This appears to me strange how different xyz files can give exactly same Sq values.
